programmers/모의고사.js

```diff
@@ -0,0 +1,16 @@
+function solution(answers) {
+  let guessPatterns = [
+    [1, 2, 3, 4, 5],
+    [2, 1, 2, 3, 2, 4, 2, 5],
+    [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
+  ];
+  let guessAnswers = guessPatterns.map((val) => {
+    return [...new Array(Math.floor(answers.length / val.length)).fill(val).flat(), ...val.slice(0, answers.length % val.length)];
+  });
+  let scores = guessAnswers.map((guessAnswer) => {
+    return guessAnswer.reduce((acc, val, idx) => acc + ((val === answers[idx]) ? 1 : 0), 0);
+  });
+  return scores.flatMap((val, idx) => (val === Math.max(...scores) ? [idx + 1] : []));
+}
```

Comment:
띄어쓰기와 줄바꿈도 컨벤션입니다.
오브첵트나 배열과 같은 Iterable 객체들은 mutable 데이터라고 해서 const로 선언해도 내부의 값을 변경할 수 있습니다.
코딩하시는 것을 보아 알고계실 듯 하지만 혹시나해서 전부다 말씀드려보는거예요 :D
메서드들을 줄지어 사용하는 것보다 나눠서 사용해주는 것이 가독성에 훨씬 좋습니다.
코딩테스트여서 이렇게 푸신거죠?ㅋㅋ

배열과 같은 iterable의 변수명들을 복수형으로 선언하시는 것 칭찬합니다~
배열 원소의 변수명을 지을 때는 val보다는 element의 약자인 el을 많이 사용한답니다.

전체적으로 상당히 어려운 메서드들을 잘 아시네요. 자스 공부를 얼마나 한건가요? (궁금)
```suggestion
function solution(answers) {
  const guessPatterns = [
    [1, 2, 3, 4, 5],
    [2, 1, 2, 3, 2, 4, 2, 5],
    [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
  ];
  
  const guessAnswers = guessPatterns.map((val) => {
    return [...new Array(Math.floor(answers.length / val.length)).fill(val).flat(), ...val.slice(0, answers.length % val.length)];
  });
  
  const scores = guessAnswers.map((guessAnswer) => {
    return guessAnswer.reduce((acc, val, idx) => acc + ((val === answers[idx]) ? 1 : 0), 0);
  });
  
  return scores.flatMap((val, idx) => (val === Math.max(...scores) ? [idx + 1] : []));
}
```

Reply:
요런식으로 반복될 패턴을 만들어줄수도 있어요 :D
```js
  const guessAnswers = [
        "12345".repeat(parseInt(answers.length / 5) + 1),
        "21232425".repeat(parseInt(answers.length / 8) + 1),
        "3311224455".repeat(parseInt(answers.length / 10) + 1)
  ];
```